### ISSUE TITLE
False positive on constructor that throws exception

### ISSUE THREAD

**Issue by ben-manes** [uber/NullAway#101]
If the construction fails with an exception then the object fields will not be set, as the object is never instantiated. However, the checker enforces that the fields are set regardless. While non-precondition exceptions from constructors are a bad idiom, sometimes oddities in 3rd party libraries have to be worked around.

```java
Foo(...) {
  Bar = ...
  try {
    this.baz = bar.baz();
  finally {
    bar.close();
  }
}
```

```
error: [NullAway] initializer method does not guarantee @NonNull fields cache, policyStats are 
initialized along all control-flow paths (remember to check for exceptions or early returns).
```

**Comment by msridhar** [comment]
We're running into this one internally as well.  The root cause is an issue with the Checker Framework dataflow library, which I think may have been fixed in their latest version.  We should take a look and see if an update fixes this problem without affecting performance too much.

**Comment by lazaroclapp** [comment]
Hey @ben-manes , we believe #123 should be a fix for this issue as well. Can you confirm?

**Comment by ben-manes** [comment]
I was able to refactor the scenario, so I am no longer effected by this. It does sound like you fixed it, so I'll close this. Thanks!
